spring-cloud-zookeeper-discovery/src/main/java/org/springframework/cloud/zookeeper/discovery/ZookeeperDiscoveryProperties.java

```diff
@@ -70,7 +79,8 @@ private ZookeeperDiscoveryProperties() {}
 
 	public ZookeeperDiscoveryProperties(InetUtils inetUtils) {
 		this.hostInfo = inetUtils.findFirstNonLoopbackHostInfo();
-		this.instanceHost = this.hostInfo.getIpAddress();
+		this.instanceHost = this.hostInfo.getHostname();
+		this.instanceIpAddress = this.hostInfo.getIpAddress();
```

Comment:
you wrote

```
IP address to use when accessing service (must also set preferIpAddress to use)
```

So either we enforce it like I suggested or we change the docs.  Also 

`return this.preferIpAddress ? this.instanceIpAddress : this.instanceHost;`

I can have `preferIpAddress == true` and an empty `this.instanceIpAddress`. So I guess you should either check in the constructor and throw an exception or do sth like this

```
public String getInstanceHost() {
 if (this.preferIpAddress && StringUtils.hasText(this.instanceIpAddress) {
return this.instanceIpAddress;
}
return this.instanceHost;
}
```